geomstats/geometry/product_riemannian_metric.py

```diff
@@ -52,21 +55,52 @@ def inner_product_matrix(self, base_point=None):
             Matrix of the inner-product at the base point.
         """
         matrix = gs.zeros([self.dimension, self.dimension])
-        b = self.dimensions[0]
-        matrix[:b, :b] = self.metrics.inner_product_matrix(base_point[0])
         dim_current = 0
-
         for i in range(self.n_metrics - 1):
             dim_current += self.dimensions[i]
             dim_next = self.dimensions[i + 1]
             a = dim_current
             b = dim_current + dim_next
-            matrix_next = self.metrics.inner_product_matrix(base_point[i + 1])
+            matrix_next = self.metrics[i].inner_product_matrix(
+                base_point[i + 1])
             matrix[a:b, a:b] = matrix_next
 
         return matrix
 
-    def inner_product(self, tangent_vec_a, tangent_vec_b, base_point=None):
+    def _detect_intrinsic_extrinsic(self, point):
```

Comment:
Ditto: `_is_intrinsic`? Although it could be nice to have this method coded only once, in product_manifold.py: how about **using** the method from product_manifold.py in this file?